src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -996,24 +1006,27 @@ export class StudyViewPageStore {
         let uniqueKey:string = getClinicalAttributeUniqueKey(chartMeta.clinicalAttribute!);
```

Comment:
we should see if we can avoid having to call this method to populate the clinicalDataCountPromises.  clinicalDataCountPromises seems like it should be computed.     if we must then method should be named something like addChartData

Reply:
This is a good point. I agree a computed map seems a fair solution. Let me reevaluate.